hw_08/task_02.js

```diff
@@ -0,0 +1,277 @@
+      };
+    })
+  })
+}
+
+structureOfCompany(enterprises);
+
+
+/*
+2. Написать функцию, которая будет принимать 1 аргумент (id отдела или название отдела и возвращать название предприятия, к которому относится).
+Пример:
+getEnterpriseName(4) // Предприятие 1
+getEnterpriseName("Отдел маркетинга") // Предприятие 2
+*/
+console.log("-------------------task_2-------------------")
+
+function getEnterpriseName(nameOrId) {
+  if (!nameOrId) {
+    return console.log("Please, provide a valid id or name")
+  }
+  for (let object of enterprises) {
+    if (object.departments.find((department) => (department.name === nameOrId || department.id === nameOrId))) {
+      return object.name;
+    };
+  }
+};
+console.log(getEnterpriseName("Отдел тестирования"));
+console.log(getEnterpriseName(7));
+
+/*
+3. Написать функцию, которая будет добавлять предприятие. В качестве аргумента принимает название предприятия
+Пример:
+addEnterprise("Название нового предприятия")
+*/
+console.log("-------------------task_3-------------------")
+
+function addNewEnterprise(name) {
+  let newDepartment = {
+    id: getLastId(),
+    name,
+    departments: null
+  };
+  return [...enterprises, newDepartment];
+}
+
+function getLastId() {
+  let id = 0;
+  enterprises.forEach((object) => {
+    if (object.id > id) {
+      id = object.id;
+    }
+    object.departments.forEach((department) => {
+      if (department.id > id) {
+        id = department.id;
+      }
+    })
+  })
+  return id += 1;
+}
+
+console.log(addNewEnterprise("Elena's department"))
+
+
+/*
+4. Написать функцию, которая будет добавлять отдел в предприятие. В качестве аргумента принимает id предприятия, в которое будет добавлен отдел и название отдела.
+Пример:
+addDepartment(1, "Название нового отдела")
+*/
+
+console.log("-------------------task_4-------------------")
+
+function addNewDepartment(id, depName) {
+  const foundedDepartment = getNewDepartmentById(id);
+  const newDepartment = createNewDepartmentByName(depName);
+  foundedDepartment.departments.push(newDepartment);
+  return foundedDepartment.departments;
+}
+
+function getNewDepartmentById(id, depName) { //найти по id департамент
+  const foundedDepartment = enterprises.find((object) => object.id === id);
+  if (!foundedDepartment) {
+    console.log(`No department found by id: ${id}`);
+  }
+  return foundedDepartment;
+}
+
+function createNewDepartmentByName(name) { //создание нового департамента
+  const newDepartment = {
+    id: getLastId(),
+    name,
+    employees_count: null,
+  }
+  return newDepartment;
+}
+
+console.log(addNewDepartment(5, "Elena's department"))
+
+/*
+5. Написать функцию для редактирования названия предприятия. Принимает в качестве аргумента id предприятия и новое имя предприятия.
+Пример:
+editEnterprise(1, "Новое название предприятия")
+*/
+console.log("-------------------task_5-------------------")
+
+function editDepartmentName(id, newDepName) {
```

Comment:
department or enterprise ?
```suggestion
function editDepartmentName(id, newDepName) {
```